File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/PostFlagConfigScheduledChangesExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PostFlagConfigScheduledChangesExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var postFlagScheduledChangesInput = new PostFlagScheduledChangesInput();
        postFlagScheduledChangesInput.executionDate(1718467200000L);
        postFlagScheduledChangesInput.instructions(JSON.deserialize("""
            [
                {
                    "kind": "turnFlagOn"
                }
            ]
        """, List.class));
        postFlagScheduledChangesInput.comment("Optional comment describing the scheduled changes");

        try
        {
            var response = new ScheduledChangesApi(config).postFlagConfigScheduledChanges(
                "projectKey_string", // projectKey
                "featureFlagKey_string", // featureFlagKey
                "environmentKey_string", // environmentKey
                postFlagScheduledChangesInput,
                null // ignoreConflicts
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling ScheduledChangesApi#postFlagConfigScheduledChanges");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/PostFlagConfigScheduledChangesExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class PostFlagConfigScheduledChangesExample
{
    fun postFlagConfigScheduledChanges()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val postFlagScheduledChangesInput = PostFlagScheduledChangesInput(
            executionDate = 1718467200000,
            instructions = Serializer.moshi.adapter<List<Map<String, Any>>>().fromJson("""
                [
                    {
                        "kind": "turnFlagOn"
                    }
                ]
            """)!!,
            comment = "Optional comment describing the scheduled changes",
        )

        try
        {
            val response = ScheduledChangesApi().postFlagConfigScheduledChanges(
                projectKey = "projectKey_string",
                featureFlagKey = "featureFlagKey_string",
                environmentKey = "environmentKey_string",
                postFlagScheduledChangesInput = postFlagScheduledChangesInput,
                ignoreConflicts = null,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling ScheduledChangesApi#postFlagConfigScheduledChanges")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling ScheduledChangesApi#postFlagConfigScheduledChanges")
            e.printStackTrace()
        }
    }
}
